### ISSUE TITLE
Need better sanity checks on DOS-style partition tables

### ISSUE THREAD

**Issue by calamares-user** [calamares/calamares#619]
Jira issue originally created by user highvoltage:

Partitions on system as we received it from user:
3 primary partitions:
* Windows system reserved (eg sda1)
* Windows system (eg sda2)
* Data (eg sda3)
1 extended partition:
* Data (eg sda5)

A normal user might want to resize their Windows or data partition to install their Linux system. In the case above, Calamares will rightly not allow them to do so (it's currently a bit crude because it doesn't offer any explanation of why it doesn't allow it, which is a seperate UX bug).

Switching to manual partitioning, a user can still manually resize the 3rd data partition (sda3) and choose to create a new partition in the newly available free space. This of course won't work, since there are already 4 primary partitions (3 primary and 1 extended), but Calamares will let the user continue with this setup and fail when it attempts to create the new partition with an error.

Ideally, Calamares should have better sanity checks in the partitioning UI and not allow the user to continue with a broken partitioning specification.

**Comment by adriaandegroot** [comment]
This was fixed in Calamares 3.2.0, as #953

**Comment by Swyter** [comment]
@adriaandegroot I've had this happen on the current version of the Manjaro installer in 2022. Have the normal three Windows 10 system partitions plus one extra NTFS one for data. All of them primary. Resize the fourth one in Gparted to leave some free space at the end. I don't remember if I already had Calamares opened in that partition step or not side by side.

In the installer it let me pick that blank space in the «replace» partition mode and keep as normal to the following steps, I fill out the user and password and click on confirming the install. When it tries to do the formatting it either failed without a message or crashed. I don't remember.

Luckily, after noticing the partition table type I remembered the limitation and deleted the less bad Windows one, allowing me to proceed after restarting/re-doing the installer. Some other user would have had a really hard time with this. I think the check logic has a few holes if this still happens. Unfortunately I can't provide more details.

PS: The Spanish translation quality in Calamares is terribly dismal; I'll look into joining the translation team and reviewing it. A good impression is important, and the users I installed it to weren't impressed.
